nomos-blend/message/src/reward/session.rs

```diff
@@ -16,11 +18,15 @@ pub struct SessionInfo {
 impl SessionInfo {
     pub fn new(
         session_number: SessionNumber,
-        session_randomness: SessionRandomness,
+        pol_epoch_nonce: ZkHash,
         num_core_nodes: u64,
-        total_core_quota: u64,
+        core_quota: u64,
         message_frequency_per_round: NonNegativeF64,
     ) -> Result<Self, Error> {
+        let total_core_quota = core_quota
+            .checked_mul(num_core_nodes)
+            .ok_or(Error::TotalCoreQuotaTooLarge(u64::MAX))?;
```

Comment:
Now I understand why you @ntn-x2 suggested https://github.com/logos-co/nomos/pull/1813#discussion_r2439901797. I realized that it's cleaner to compute the `total_core_quota` here inside `SessionInfo` because all other code deals with only `core_quota`. Also, it's okay that this module knows how the `total_core_quota` is computed, because its definition is already implied in the way calculating the activity threshold.

Reply:
Yes the concept of total core quota is actually only used for rewards, so I think it's fair to let this bit of code know how to compute that.